### PR TITLE
[GEOS-9924] Promoting MapML community module to extension

### PR DESCRIPTION
**Pull request by prushforth**
This PR promotes the MapML community module to extension.  Test coverage has recently been improved to the required level, although more will be done.  Almost all the steering committee has [voted](https://github.com/geoserver/geoserver/wiki/GSIP-200#voting).  The required time hasn't elapsed, but there is an upcoming release so it would be ideal to get into that.  Please let me know if there are other things that must be done, (assuming all the required checks pass).

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a [ticket](https://osgeo-org.atlassian.net/browse/GEOS-9924?atlOrigin=eyJpIjoiNzQ2MDY2OGE3MDQ2NDU5ZDk0NmIwMDBjZGE3Y2YzODEiLCJwIjoiaiJ9) in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: doc/en/user/source/extensions/mapml/index.rst [hunk 1]
```diff
@@ -3,20 +3,22 @@
...
 
 The MapML module for GeoServer adds new MapML resources to access WMS and WFS services configured in Geoserver. The MapML modules includes support for styles, tiling, querying, sharding, and dimensions options for WMS layers, and also provides a MapML outputFormat for WMS GetFeatureInfo and WFS GetFeatures requests. See below for information on installing and configuring the MapML module.
 
+    .. warning:: MapML is an experimental proposed extension of HTML for the Web. The objective of the format is to standardize map, layer and feature semantics in HTML.  As the format progresses towards a Web standard, it may change.  Always use the latest version this extension, and follow the project's progress at https://maps4html.org.
```
Comment (bradh) [inline doc/en/user/source/extensions/mapml/index.rst:10]: ...latest version *of* this extension....

File: doc/en/user/source/extensions/mapml/index.rst [hunk 2]
```diff
@@ -122,37 +124,26 @@ With the MapML Community Module installed, the GeoServer Layer Preview page is m
...
 
-The only tool which is presently able to display MapML is a Leaflet-based MapML client. You can get your own copy of the client by using the npm "bower" package management tool.  Once you have bower installed, you can install the web-map client in a directory of your choice, by running the "bower install web-map" command in that directory. This will create a "bower_components" directory in the directory in which you execute the command. This client can be imported into an HTML page with the appropriate ``<map>`` and ``<layer>`` elements to reference the MapML resources defined above. Here is a simple, self-contained example of such an HTML page: 
+MapML visualization is supported by the Web-Map-Custom-Element project. The MapML viewer is built into the GeoServer layer and layer group preview facility.  You can find out more about the Web-Map-Custom-Element at the project `web site <https://maps4html.org/web-map-doc/>`. Here is a simple, self-contained example an HTML page that uses the <mapml-viewer> and <layer-> elements:
```
Comment (bradh) [inline doc/en/user/source/extensions/mapml/index.rst:127]: example *of a* HTML page?

File: doc/en/user/source/extensions/mapml/index.rst (hunk 1 above)
Comment (prushforth) [reply doc/en/user/source/extensions/mapml/index.rst]: Fixed, thanks!

**Review by aaime — COMMENTED** [review]
Overall it looks good. I don't see changes in ``web/app/pom.xml``, so assuming it's still pointing to the community jar?

```
    <profile>
      <id>mapml</id>
      <dependencies>
        <dependency>
          <groupId>org.geoserver.community</groupId>
          <artifactId>gs-mapml</artifactId>
          <version>${project.version}</version>
        </dependency>
      </dependencies>
    </profile>
```

**Comment by prushforth** [comment]
> Overall it looks good. I don't see changes in `web/app/pom.xml`, so assuming it's still pointing to the community jar?

Thanks I will fix that.

**Comment by aaime** [comment]
Aaand merged!

**Comment by prushforth** [comment]
Awesome!  Thanks for your help @aaime !

**Pull request merged**